Utilities/File.cpp

```diff
@@ -1898,13 +1903,72 @@ bool fs::file::strict_read_check(u64 _size, u64 type_size) const
 	return true;
 }
 
+std::string fs::get_executable_path()
+{
+#if defined(_WIN32)
+	wchar_t buffer[32767];
+	GetModuleFileNameW(nullptr, buffer, sizeof(buffer) / 2);
+	return wchar_to_utf8(buffer);
+#elif defined(__APPLE__)
+	char bin_path[PATH_MAX];
+	uint32_t bin_path_size = sizeof(bin_path);
+	if (_NSGetExecutablePath(bin_path, &bin_path_size) != 0)
+	{
+		sys_log.error("Failed to find app binary path");
+		return {};
+	}
+
+	return std::filesystem::path(bin_path).parent_path().parent_path().parent_path().string();
```

Comment:
Avoid using std::filesystem please, we had our issues with it in the past with some unicode path characters. You can use fs::get_parent_dir and fs::is_dir for example. I know that the original code had mistaken here,